gulpfile.js

```diff
@@ -0,0 +1,381 @@
+'use strict';
+/**
+ * This file needs to be JavaScript and is read by gulp.
+ */
+// Global imports.
+const child_process = require('child_process');
+const fs = require('fs');
+const gulp = require('gulp');
+const path = require('path');
+const resolveBin = require('resolve-bin');
+
+// Other imports.
+const inlineResources = require('./scripts/release/inline-resources');
+const karma = require('karma');
+
+// Gulp plugins.
+const gulpClean = require('gulp-clean');
+const gulpServer = require('gulp-server-livereload');
+const gulpMerge = require('merge2');
+const gulpRunSequence = require('run-sequence');
+const gulpSass = require('gulp-sass');
+const gulpSourcemaps = require('gulp-sourcemaps');
+const gulpTs = require('gulp-typescript');
+
+
+// Directories.
+const srcDir = path.join(__dirname, 'src');
+const componentsDir = path.join(srcDir, 'lib');
+const devAppDir = path.join(srcDir, 'demo-app');
+const e2eAppDir = path.join(srcDir, 'e2e-app');
+
+const outDir = 'dist';
+const outLibDir = path.join(outDir, '@angular2-material');
+
+
+/**
+ * Create a TS Build Task, based on the options.
+ */
+function createTsBuildTask(options) {
```

Comment:
Is `options` of any particular type? I generally try to avoid JS functions that just accept some `options` object without any additional information about what the options are. Makes me wonder if we could have this as a `gulpfile.ts` instead of `.js`.